guides/micronaut-microservices-distributed-tracing-zipkin-opentelemetry/micronaut-microservices-distributed-tracing-zipkin-opentelemetry.adoc

```diff
@@ -0,0 +1,182 @@
+common:header-top.adoc[]
+
+== Getting Started
+
+In this guide, we will integrate https://zipkin.io[Zipkin] in a Micronaut application composed of three microservices.
+
+____
+Zipkin is a distributed tracing system. It helps gather timing data needed to troubleshoot latency problems in microservice architectures. It manages both the collection and lookup of this data.
+____
+
+You will discover how the Micronaut framework eases Zipkin integration.
+
+common:requirements.adoc[]
+
+common:completesolution.adoc[]
+
+== Writing the Application
+
+To learn more about this sample application, read the guideLink:micronaut-microservices-services-discover-consul[Consul and the Micronaut Framework - Microservices Service Discovery] guide. The application contains three microservices.
+
+* `bookcatalogue` - This returns a list of books. It uses a domain consisting of a book name and an ISBN.
+
+* `bookinventory` - This exposes an endpoint to check whether a book has sufficient stock to fulfil an order. It uses a domain consisting of a stock level and an ISBN.
+
+* `bookrecommendation` - This consumes previous services and exposes an endpoint that recommends book names that are in stock.
+
+The `bookrecommendation` service consumes endpoints exposed by the other services. The following image illustrates the application flow:
+
+image::flow.svg[]
+
+A request to `bookrecommendation` (http://localhost:8080/books) triggers several requests through our microservices mesh.
+
+common:annotationprocessors.adoc[]
+
+== Zipkin and the Micronaut Framework
+
+=== Install Zipkin via Docker
+
+The quickest way to start https://zipkin.io[Zipkin] is via Docker:
+
+[source,bash]
+----
+docker run -d -p 9411:9411 openzipkin/zipkin
+----
+
+
+== OpenTelemetry dependencies
+
+common:opentelemetry-annotations.adoc[]
+
+common:opentelemetry-http.adoc[]
+
+common:opentelemetry-exporter.adoc[]
+
+Each service adds the OpenTelemetry Zipkin exporter dependency.
+
+dependency:opentelemetry-exporter-zipkin[groupId=io.opentelemetry,scope=compile]
+
+Each service sets `zipkin` as exporter in configuration:
+
+resource:application.yml[app=bookcatalogue,tag=zipkin]
+
+common:opentelemetry-disable.adoc[]
+
+=== Changes Book inventory
```

Comment:
Should this be **Change** Book Inventory?

Reply:
I think it should be

> Changes **to** Book inventory

So I've changed it to that